src/question_matrixdropdownrendered.ts

```diff
@@ -411,17 +411,18 @@ export class QuestionMatrixDropdownRenderedTable extends Base {
   private hasActionCellInRowsValues: any = {};
   private hasActionCellInRows(location: "start" | "end"): boolean {
     if (this.hasActionCellInRowsValues[location] === undefined) {
-      var rows = this.matrix.visibleRows;
-      this.hasActionCellInRowsValues[location] = false;
-      for (var i = 0; i < rows.length; i++) {
-        if (!this.isValueEmpty(this.getRowActions(i, location))) {
-          this.hasActionCellInRowsValues[location] = true;
-          break;
-        }
-      }
+      this.hasActionCellInRowsValues[location] = this.hasActionsCellInLocaltion(location);
     }
     return this.hasActionCellInRowsValues[location];
   }
+  private hasActionsCellInLocaltion(location: "start" | "end"): boolean {
+    if(location == "end" && this.hasRemoveRows) return true;
+    var rows = this.matrix.visibleRows;

```

Comment:
small refactoring: replace 420-424 lines with one
return this.matrix.visibleRows.some((row, index) => !this.isValueEmpty(this.getRowActions(index, location)));